presto-main/src/test/java/io/prestosql/operator/TestingExchangeHttpClientHandler.java

```diff
@@ -73,13 +74,18 @@ public Response handle(Request request)
             headers.put(PRESTO_PAGE_NEXT_TOKEN, String.valueOf(pageToken + 1));
             headers.put(PRESTO_BUFFER_COMPLETE, String.valueOf(false));
             DynamicSliceOutput output = new DynamicSliceOutput(256);
+            output.writeInt(SERIALIZED_PAGES_MAGIC);
```

Comment:
Nit: Maybe extract method `writeSerializedPagesHeader(int numPages)` and put it in `PagesSerdeUtil`?
Or maybe MAGIC and pages count should be written by `writeSerializedPages`?

Reply:
Extracting write is nice, but it would call for extracting read as well. And this won't work, unless I introduce a new class for the header (count, magic).
It would be warranted if we wanted to pack more information here, or make the protocol extensible / future-proof, but since we don't have such plans, let me not do this now.